baybe/utils/dataframe.py

```diff
@@ -413,3 +414,42 @@ def fuzzy_row_match(
             inds_matched.extend(inds_found)
 
     return pd.Index(inds_matched)
+
+
+def pretty_printing_dataFrame(df: pd.DataFrame) -> str:
+    """Return a pretyt/readable str Object.
+
+    Args:
+        df: The dataframe to be printed.
+
+    Returns:
+        The values to be printed as a str table.
+    """
+    # select the first 10 rows/values and store them in a str table
+    # TODO: remove hard coded numbers and allow user defined specifications
+
+    str_df = ""
+
+    # Handling with Empty dataFrame
+    if len(df) == 0:
+        return "This attribute is empty. "
+
+    # Extract extra columns
+    if len(df.columns) > 5:
+        df = df.T.head(5).T
+        str_df = (
+            "\n Your dataFrame has too many columns. "
+            + "For a better readable table only the first 5 columns will be printed. \n"
```

Comment:
I'm not so happy with these messages, which will just flood the output with text. Instead, I'd very much prefer a "dotted" visualization (similar to what pandas would to if there are too many rows/columns)

Reply:
Let me use this thread to open a general discussion: 
1. I guess we agree that "pretty printing" a dataframe means to just print a reduced number of rows and columns. Do we want to have these numbers fixed, and if so, which number should we choose? I think the choices made by @RimRihana  here are reasonable as defaults, and I am having trouble imagining a good way how a user could specify it.
2. Should we offer the possibility to turn off pretty printing? If so, how could we do that? This feels like either something that could be configured via an environment variable or maybe by a hidden attribute of the campaign class, since I would imagine that a user would not want to enable pretty print only for individual aspects.

Reply:
Regarding informative text: If we decide to only print a fraction of the output, we should inform the user about that. I'd however be fine if we either shorten the text (maybe like `Printing first 10/123 rows and 5/12 columns`) or maybe add that smartly into the table (e.g. by showing first 4 rows, one rows with dots and then the final row including the row number)

Reply:
1. and 2.: Don't worry about user customization for now. If the user wants the full info, they can always go via the `__repr__` still. And while we can indeed make things configurable at some point, this would rather be a topic for the general baybe `Settings` that I have on my bucket list since ages. It would not go via class attributes, but rather via a dedicated settings mechanism, probably some mix of how `gpytorch` and `foundry-dev-tools` do it (i.e., I imagine something similar to an attrs singleton/borg class controlling baybe behavior, defining defaults, and validating user adjustments.

3. Yes, let's have a look at the screenshots. But the simplest thing I have in mind is
   *  Printing the dataframe in one single "line", i.e. not having a line break after a certain number of columns
   * Showing dots if stuff gets too big.
   
Our main goal we want to achieve here is that the output is easy to parse for the user and the shape does not depend on the content (i.e., currently the number of lines you need to scroll to get to the next attributes depends on how big the dataframes are, and there is no visual boundary that separates stuff)